custom_components/mikrotik_router/mikrotik_controller.py

```diff
@@ -2015,7 +2014,128 @@ def _address_part_of_local_network(self, address):
     #   _get_accounting_uid_by_ip
     # ---------------------------
     def _get_accounting_uid_by_ip(self, requested_ip):
-        for mac, vals in self.data["accounting"].items():
+        for mac, vals in self.data["client_traffic"].items():
+            if vals.get("address") is requested_ip:
+                return mac
+        return None
+
+    # ---------------------------
+    #   _get_iface_from_entry
+    # ---------------------------
+    def _get_iface_from_entry(self, entry):
+        """Get interface default-name using name from interface dict"""
+        uid = None
+        for ifacename in self.data["interface"]:
+            if self.data["interface"][ifacename]["name"] == entry["interface"]:
+                uid = ifacename
+                break
+
+        return uid
+
+    # ---------------------------
+    #   process_kid_control
+    # ---------------------------
+    def process_kid_control_devices(self):
+        """Get Kid Control Device data from Mikrotik"""
+
+        uom_type, uom_div = self._get_unit_of_measurement()
+
+        # Build missing hosts from main hosts dict
+        for uid, vals in self.data["host"].items():
+            if uid not in self.data["client_traffic"]:
+                self.data["client_traffic"][uid] = {
+                    "address": vals["address"],
+                    "mac-address": vals["mac-address"],
+                    "host-name": vals["host-name"],
+                    "previous-bytes-up": 0.0,
+                    "previous-bytes-down": 0.0,
+                    "wan-tx": 0.0,
+                    "wan-rx": 0.0,
+                    "tx-rx-attr": uom_type,
+                    "available": False,
+                    "local_accounting": False,
+                }
+
+        _LOGGER.debug(f"Working with {len(self.data['client_traffic'])} kid control devices")
+
+        time_diff = self.api.take_client_traffic_snapshot(False)
+        if not time_diff:
+            return
+
+        kid_control_devices_data = parse_api(
+            data={},
+            source=self.api.path("/ip/kid-control/device"),
+            key="mac-address",
+            vals=[
+                {"name": "mac-address"},
+                {"name": "bytes-down"},
+                {"name": "bytes-up"},
+                {
+                    "name": "enabled",
+                    "source": "disabled",
+                    "type": "bool",
+                    "reverse": True,
+                }
+            ]
+        )
+
+        if not kid_control_devices_data:
+            _LOGGER.debug("No kid control devices found, make sure kid-control feature is configured")
```

Comment:
This should be handled in same way as duplicate. Reported as Error so people see it and shown only once.

Reply:
Hmm, is this a good idea? This will require of user to restart homeassistant after enabling kid-control feature to start polling data again.
Maybe write .error one time so user see it in log but after that we should poll data anyhow, in case user enables kid-control feature?

Reply:
Yes, thats what I mean. Just error message once for user information and prevent log spam.
Functionality itself should not break.